src/runtime/hexagon/hexagon_buffer.cc

```diff
@@ -57,17 +57,30 @@ struct DDRAllocation : public Allocation {
 
 struct VTCMAllocation : public Allocation {
   VTCMAllocation(size_t nbytes, size_t alignment) : Allocation(nbytes, alignment) {
-    // TODO(HWE): Handle alignments greater than 2k
-    CHECK(alignment <= 0x800) << "VTCMAllocation called for invalid alignment";
-    if ((nbytes & 0x7FF) && ((alignment & 0x7FF) == 0)) {
-      // Caller has requested 2k alignment, but the size is not a multiple of 2k
-      // Adjust size to be a multiple of 2k so that we will allocate from the front of the pool
+    // For simplicity, the current VTCM dynamic pool supports the following alignments: less than
+    // or equal to 128 (0x80), and 2k (0x800)
+    CHECK((alignment <= 0x80) || (alignment == 0x800))
+        << "VTCMAllocation called for invalid alignment " << alignment;
+
+    if ((alignment == 0x800) && (nbytes & 0x7FF)) {
+      // Caller has requested 2k alignment, but the size is not a multiple of 2k.
+      // Adjust size to be a multiple of 2k so that we will allocate from the front of the pool.
       nbytes = nbytes >> 11;
       nbytes = nbytes << 11;
       nbytes += 0x800;
```

Comment:
Side note---an alternative would be `nbytes = (nbytes + 0x7ff) & -0x800`, and it would work without the check if `nbytes & 0x7ff`.  Same below.